### ISSUE TITLE
Server list is reset after scanning Central QR code

### ISSUE THREAD

**Issue by kkrawczyk123** [getodk/collect#4093]
#### Software and hardware versions 
Collect v1.28.x, Android v all  device used...

#### Problem description
Previously used server URL should be stored in dropdown list and gives user chance to go back to already used ones - especially Centrals as they are hard to input manually. 
![ezgif com-video-to-gif (31)](https://user-images.githubusercontent.com/33342488/93238148-5230e780-f781-11ea-8814-c74a68d8c16f.gif)

#### Steps to reproduce the problem
1. Use aggregate server other than default one
2. Scan any Central QR code: only default one is visible on dropdown list now
3. Go back to default server: Central URL is also gone from dropdown list

#### Expected behavior
URLs should be stored on the dropdown list up to 5 URLs: original PR: https://github.com/getodk/collect/pull/3352
screenshot from 1.27:
![Screenshot (15 Sep 2020 18_16_33)](https://user-images.githubusercontent.com/33342488/93236951-bce12380-f77f-11ea-892f-9d6e90e8f179.png)

#### Other information 
Things you tried, stack traces, related issues, suggestions on how to fix it...
@getodk-bot label "regression"

**Comment by grzesiek2010** [comment]
I have a problem with this issue... it was surprising to me that we reset all settings before we load the new ones.
I needed to ask @seadowg and @kkrawczyk123 for confirmation if it's not a bug they said it's intentional but at least @seadowg was a bit confused as well right?

What if we have a user who has their own settings and use qr codes just to load long urls from central? After scanning such a qr code everything else disappears? I think it's not ok.

On the other hand if you are somehow able to convince my that the behavior is correct then if we reset completly everything we should reset the history of previously used urls as well and we could close this issue.

@lognaturel @seadowg help!

**Comment by seadowg** [comment]
> What if we have a user who has their own settings and use qr codes just to load long urls from central? After scanning such a qr code everything else disappears? I think it's not ok.

@grzesiek2010 yeah the behaviour definitely doesn't work in this case. I've got a card (https://github.com/getodk/collect/projects/6#card-44019168) to start investigate this as part of the "Onboarding" project. In brief discussion @lognaturel and I have had about it the idea that's been floated is having QR codes have a `reset_to_default` key that signals to Collect whether to reset everything else. That way Central could have that as `false` and just URL would be imported but device to device settings imports (where you might want to reset to defaults) could still happen. That's just an idea and of course we'll discuss when we get to it!

> On the other hand if you are somehow able to convince my that the behavior is correct then if we reset completly everything we should reset the history of previously used urls as well and we could close this issue.

Hmmm I'm not sure. My one worry is that the sense I've gotten (I'd like to do user interviews around this) is that people do switch between servers and that actually they'd like the server list to be enhanced (an example forum post is [here](https://forum.getodk.org/t/server-url-shown-on-home-screen-multiple-servers-switch-with-colour-change/23340)). Given that I'm worried we've made these workflows even harder for people. Maybe for the moment we should maintain this behaviour and store the `KNOWN_URL_LIST` in meta shared preferences? One potential problem with that is that it would only be "resettable" using Android's settings. Maybe that's OK though?
